Implement OfflineMode tests

PhenX opened the pull request:
closes #7

PhenX commented:
Yes "it works locally" ©, but not on AppVeyor :(

kblok commented:
Yeah... I'm getting that error locally (on Mac) if I run all the tests though.

PhenX commented:
I'll give it a look. I think it might be due to the fact that the same Page instance is used between tests, but I'll have to confirm.

kblok commented:
I tried creating a new Page and even a new Browser in that test and now joy, but give it a shot.

Meir017 commented:
each test gets a new Browser and a new Page, they are not reused between tests

PhenX commented:
@Meir017 that's not true if test is run among a collection (https://xunit.github.io/docs/shared-context#collection-fixture), which it was (and other tests are like this too).
I removed the collection attribute and it seems fine locally, but not on AppVeyor, maybe because it's too much isolated from other tests... I think we should include executable download/check in the initialize method ... or find something else.

Meir017 commented:
hmm, I guess we should change it so each test gets a new instance of `Browser`.
I'm not totally sure what to do, not very familiar with xunit (I'm used to MSTest...).
Do you know what needs to change?

PhenX commented:
🤬

kblok commented:
@PhenX let's try to reduce the PR scope. We don't need to clean up all those usings right now.

kblok commented on lib/PuppeteerSharp.TestServer/PuppeteerSharp.TestServer.csproj:
```diff
@@ -1,13 +1,11 @@
-<Project Sdk="Microsoft.NET.Sdk.Web">
+﻿<Project Sdk="Microsoft.NET.Sdk.Web">
```
Let's undo this change. It makes no difference

PhenX replied:
Yeah, it's Rider that makes some formatting ...

PhenX replied:
I think it's better now

PhenX replied:
I can't see the difference on this line, maybe the BOM or different carriage return on that line ...

kblok commented:
Let me know if you need a hand here @PhenX

PhenX commented:
@kblok yes, I think I need help here ... I changed the way fixtures are implemented, to follow async pattern (with the `IAsyncLifetime` interface) because I thought it was this, and I checked that Browser and Page were always renewed, and that's the case, and as a last resot I updated Nuget packages.
Still no luck on AppVeyor, no Exception is thrown when in offline mode, while I always have one locally, wether I run only one or all of them ...

kblok commented:
No worries @PhenX. I'll take a look at this code tomorrow. Again, having tests coded is super helpful, this web sockets communication with chromium can be quite tricky.

In the meantime, if you want, you can take this guy #13. The implementation should be quite similar to the Dialog feature https://github.com/kblok/puppeteer-sharp/pull/120.

kblok commented:
I'm digging into this guy, it's related to the same error we're getting with @Meir017 here https://github.com/kblok/puppeteer-sharp/pull/110#issuecomment-380797643

We're not letting `Navigate` fail.

kblok commented:
How are we going with this guy @PhenX? I think if you pull from master this will improve.

kblok commented:
Continues on #150
